### ISSUE TITLE
[BUG] Classname `peer` from tailwindcss is being reported by `no-custom-classname` rule

### ISSUE THREAD

**Issue by robert-barahona** [francoismassart/eslint-plugin-tailwindcss#459]
**Describe the bug**
`peer` is being reported by `no-custom-classname` rule.

**To Reproduce**

1. Go to this [repo](https://github.com/robert-barahona/next-app-2)
2. Install deps `pnpm install`
3. Run `pnpm eslint`

**Expected behavior**
`peer` shouldn't be reported

**Screenshots**
<img width="862" height="199" alt="Image" src="https://github.com/user-attachments/assets/bdf7a342-a8a3-422d-aaa1-433e3507e5b7" />

**Environment (please complete the following information):**

- OS: macOs
- Softwares + version used:
  - eslint 9
  - tailwindcss 4
  - eslint-plugin-tailwindcss 4.0.3

**Additional context**

`peer` is a classname from [tailwind](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-the-descendants-of-a-peer)

```jsx
    <div className="m-10">
      <label className="peer select-none">
        <input className="me-2" type="checkbox" name="todo[1]" />
        Hide text bellow
      </label>
      <p className="mt-4 peer-has-checked:hidden">Hello World!</p>
    </div>
```

**eslint config file or live demo**
https://github.com/robert-barahona/next-app-2

**Comment by francoismassart** [comment]
Hi, I've met the issue too, it is fixed in v4.0.4 which is already released
